src/main/java/spleetwaise/transaction/logic/commands/AddCategoryCommand.java

```diff
@@ -0,0 +1,74 @@
+package spleetwaise.transaction.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static spleetwaise.transaction.logic.parser.CliSyntax.PREFIX_CATEGORY;
+import static spleetwaise.transaction.logic.parser.CliSyntax.PREFIX_TXN;
+
+import spleetwaise.commons.logic.commands.Command;
+import spleetwaise.commons.logic.commands.CommandResult;
+import spleetwaise.commons.logic.commands.exceptions.CommandException;
+import spleetwaise.transaction.model.transaction.Transaction;
+
+/**
+ * This class represents a command for adding categories to transactions.
+ */
+public class AddCategoryCommand extends Command {
+
+    /**
+     * The word used to trigger this command in the input.
+     */
+    public static final String COMMAND_WORD = "addTxnCat";
+
+    public static final String MESSAGE_SUCCESS = "Category added to transaction: [%s] with [%s]";
+
+    public static final String MESSAGE_EMPTY_CATEGORY = "Cannot define empty category.";
+
+    /**
+     * The message usage string that explains how to use this command.
+     */
+    public static final String MESSAGE_USAGE =
+            COMMAND_WORD + ": Add a category to transaction.\n" + "Parameters: " + PREFIX_TXN + "TRANSACTION_INDEX "
+                    + PREFIX_CATEGORY + "CATEGORY " + "Example: " + COMMAND_WORD + " " + PREFIX_TXN + "1 "
+                    + PREFIX_CATEGORY + "Transport";
+
+    private final Transaction transaction;
+    private final String category;
+
+    /**
+     * Creates an AddCategoryCommand to add tag to specified {@code Transaction}.
+     *
+     * @param transaction The transaction where the category is added.
+     * @param category The category to be added
+     */
+    public AddCategoryCommand(Transaction transaction, String category) {
+        requireNonNull(transaction);
+        requireNonNull(category);
+
+        this.transaction = transaction;
+        this.category = category;
+    }
+
+    @Override
+    public CommandResult execute() throws CommandException {
+        if (category.length() <= 0) {
+            throw new CommandException(MESSAGE_EMPTY_CATEGORY);
+        }
+
+        transaction.addCategory(category);
```

Comment:
hmm actually, are you sure this line is doing anything? I traced the `transaction` reference and it appears to be passed from a `ReadOnlyTransactionBook`.

Doing this is dangerous because `ReadOnlyTransactionBook` is supposed to be read-only. This causes the version of the transaction data in-memory to diverge from what is on the disk (JSON file), which might cause problems with subsequent commands that operate on transactions as well.